src/test/java/com/github/dockerjava/cmd/PushImageCmdIT.java

```diff
@@ -61,19 +56,20 @@ public void pushLatest() throws Exception {
     @Test
     public void pushNonExistentImage() throws Exception {
 
-        if (getVersion(dockerRule.getClient())
+        /*if (getVersion(dockerRule.getClient())
                 .isGreaterOrEqual(RemoteApiVersion.VERSION_1_26)) {
          // no errors??
         } else if (getVersion(dockerRule.getClient())
                 .isGreaterOrEqual(RemoteApiVersion.VERSION_1_24)) {
             exception.expect(DockerClientException.class);
         } else {
             exception.expect(NotFoundException.class);
-        }
+        }*/
```

Comment:
you will have different errors for different api versions, i spent a lot of time in this place

Reply:
Yes true, but it seems that only swarm behaves differently here compared to docker. The main error here was that awaitCompletion and not awaitSuccess was called. 
The error handling for was only in awaitSuccess. I have moved that to awaitCompletion since you will expect correct error handling there.